dpytools/config/properties/intproperty.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional
+from dataclasses import dataclass
+from .base import BaseProperty
+
+@dataclass
+class IntegerProperty(BaseProperty):
+    min_val: Optional[int]
+    max_val: Optional[int]
+
+    def type_is_valid(self):
+        """
+        Validate that the property looks like
+        its of the correct type 
+        """
+        try:
+            int(self.value)
+        except Exception as err:
+            raise Exception(f"Cannot cast {self.name} value {self.value} to integer.") from err
+
+    def secondary_validation_passed(self):
```

Comment:
either just `secondary_validation` or change the method on the base class to `secondary_validation_passed` please.

can you check the other one of these as well please.